x-pack/legacy/plugins/ml/public/application/management/index.ts

```diff
@@ -35,7 +35,9 @@ const licencingSubscription = plugins.licensing.license$.subscribe(license => {
   if (license.check(PLUGIN_ID, MINIMUM_FULL_LICENSE).state === LICENSE_CHECK_STATE.Valid) {
     initManagementSection();
     // unsubscribe, we only want to register the plugin once.
-    licencingSubscription.unsubscribe();
+    if (licencingSubscription !== undefined) {
```

Comment:
Is it a valid expectation for the `licensingSubscription` observable to end up being `undefined`? If it's undefined because of some other error higher up, we log the error where it's happening. It feels like this is a side-effect of something else, but we don't know what.

Reply:
From looking at observable docs - looks like unsubscribing within a subscribe callback in an observable is not the recommended way of doing it. I'd imagine that it is possible since the subscription wouldn't have returned yet when this is first hit - so it could be `undefined`. 

I've confirmed that this works so it should unblock us. I'm working on a more rxjs-ey solution but didn't want to keep everyone waiting while I tested.

Reply:
it's also worth noting that this is legacy code that will be rewritten soon for NP.
License checks where added here to enable ML's management plugin for the time being.

Reply:
Thanks for the confirmation that there will be future work done to improve this part of the code. I'm OK with the changes as they are to get folks unblocked.